sdk/core/azure-core/src/main/java/com/azure/core/util/logging/LogLevel.java

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.util.logging;
+
+import java.util.Arrays;
+import java.util.Locale;
+import java.util.Map;
+import java.util.stream.Collectors;
+import reactor.util.function.Tuple2;
+import reactor.util.function.Tuples;
+
+/**
+ * Enum which represent logging levels used in Azure SDKs.
+ */
+public enum LogLevel {
+    /**
+     * Indicates that log level is at verbose level.
+     */
+    VERBOSE("1", "verbose"),
+
+    /**
+     * Indicates that log level is at information level.
+     */
+    INFORMATIONAL("2", "info", "information", "informational"),
+
+    /**
+     * Indicates that log level is at warning level.
+     */
+    WARNING("3", "warn", "warning"),
+
+    /**
+     * Indicates that log level is at error level.
+     */
+    ERROR("4", "err", "error");
+
+    private final String[] allowedLogLevelVariables;
+
+    LogLevel(String... allowedLogLevelVariables) {
+        this.allowedLogLevelVariables = allowedLogLevelVariables;
+    }
+
+    /**
+     * Converts the log level into a numeric representation used for comparisons.
+     *
+     * @return The numeric representation of the log level.
+     */
+    public int getLogLevel() {
+        return Integer.parseInt(allowedLogLevelVariables[0]);
```

Comment:
This isn't good, as you're now doing this parse every time we need to know the numeric value, which is very frequently. You should do what Srikanta proposed previously and have the int as well as the string values.